v2/examples/spiffe-jwt-using-proxy/README.md

```diff
@@ -0,0 +1,224 @@
+- socketPath is the address of the Workload API (`unix:///tmp/agent.sock`) to which the internal Workload API client connects to get up-to-date SVIDs. Alternatively, we could have omitted this configuration option, in which case the listener would have used the `SPIFFE_ENDPOINT_SOCKET` environment variable to locate the Workload API. The code could have then been written like this:
+```go
+	jwtSource, err := workloadapi.NewJWTSource(ctx)
+```
+Althought both workloads create a `JWTSource`, it is used differently in each case:
+
+The **client workload** uses the `JWTSource` to get a JWT-SVID by calling its [FetchJWTSVID](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/workloadapi?tab=doc#JWTSource.FetchJWTSVID) function:
+```go
+	svid, err := jwtSource.FetchJWTSVID(ctx, jwtsvid.Params{
+		Audience: audience,
+	})
+```
+Where:
+- ctx is a `context.Context`. `FetchJWTSVID` blocks until a response is received or this context times out or is cancelled.
+- audience is the intended recipient of the JWT-SVID, by default it is `spiffe://example.org/server`, otherwise it is equals to the value passed as the first argument of the client's executable.
+
+Then, the client uses the JWT-SVID to set a bearer token to the request's `Authorization` header:
+```go
+	req, err := http.NewRequest("GET", serverURL, nil)
+	.
+	.
+	.
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", svid.Marshal()))
+```
+
+The **server workload** uses the `JWTSource` to authenticate the client by calling the [ParseAndValidate](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/svid/jwtsvid?tab=doc#ParseAndValidate) function:
+```go
+	_, err := jwtsvid.ParseAndValidate(token, a.jwtSource, a.audiences)
+```
+Where:
+- token is the marshalled JWT-SVID sent by the client in the `Authorization` header.
+- a.jwtSource is the `JWTSource`.
+- a.audiences is a slice of strings. Specifies a list of expected audiences in the `aud` field of the token.
+
+When `ParseAndValidate` returns an error, the server returns an `Unauthorized` status. Otherwise, the request continues normal processing.
+
+## That is it!
+As we can see the go-spiffe library allows your application avoiding to deal with the implementation details of the Workload API. You just create the SVID sources you need and then ask the library for what you need in an ergonomic fashion.
+
+## Building
+To build the client workload:
+```bash
+cd examples/spiffe-jwt-using-proxy/client
+go build
+```
+
+To build the proxy workload:
+```bash
+cd examples/spiffe-jwt-using-proxy/proxy
+go build
+```
+
+To build the server workload:
+```bash
+cd examples/spiffe-jwt-using-proxy/server
+go build
+```
+
+## Running
+This example assumes the following preconditions:
+- There is a SPIRE server and agent up and running.
+- There is a Unix workload attestor configured.
+- The trust domain is `example.org`.
+- The agent SPIFFE ID is `spiffe://example.org/host`.
+- There is a `server-workload` and `client-workload` user in the system.
```

Comment:
```suggestion
- There are `server-workload` and `client-workload` users in the system.
```
Does the group of these two users need to be documented? Do the executables need to be chowned somehow?

Reply:
No need to document the group, because it is not used when creating the registration entries.
About the executables permissions...I didn't need to chown them when testing the scenario...so I would say they don't need to be chowned.